homeassistant/helpers/entity_registry.py

```diff
@@ -394,8 +394,8 @@ def async_is_registered(self, entity_id: str) -> bool:
 
     @callback
     def async_get(self, entity_id: str) -> RegistryEntry | None:
-        """Get EntityEntry for an entity_id."""
-        return self.entities.get(entity_id)
+        """Get EntityEntry for an entity_id or entity entry id."""
+        return self.entities.get(entity_id) or self.entities.get_entry(entity_id)
```

Comment:
Do we really want to make this change just for alarm control panel?

Reply:
This does not benefit only alarm control panel, it's needed to ensure a device automation can have either an entity_id (legacy) or entity uuid (newly created).